### PR TITLE
refactor geocoding to make specific api-geocoder a sub-class of base …

### PR DESCRIPTION
**Pull request by sloypog**
…geocoder

### CATEGORY

Choose one

- [ ] Bug Fix
- [ ] Enhancement (new features, refinement)
- [x] Refactor
- [ ] Add tests
- [ ] Build / Development Environment
- [ ] Documentation

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/utils/geocoders.py [hunk 1]
```diff
@@ -14,37 +14,54 @@
...
 
     logger = logging.getLogger(__name__)
     interruptflag = False
     conf: dict = {}
     progress: dict = {}
 
-    def __init__(self, config):
+    def __init__(self, config: dict):
         self.conf = config
+        self._set_initialstates()
```
Comment (michellekunz) [inline superset/utils/geocoders.py]: initial_states

File: superset/utils/geocoders.py [hunk 2]
```diff
@@ -14,37 +14,54 @@
...
-            if geocoder == "maptiler":
-                return self._geocode_maptiler(data)
-            return self._geocode_testing()
-        finally:
-            self.progress["progress"] = 0
-            self.progress["is_in_progress"] = False
+    def geocode(self, data: list):
+        raise NotImplementedError
+
+    def _append_cords_to_dataentry(self, datum: list, geocoded: list):
```
Comment (michellekunz) [inline superset/utils/geocoders.py]: data_entry

File: superset/utils/geocoders.py [hunk 3]
```diff
@@ -56,95 +73,64 @@ def _geocode_maptiler(self, data: list) -> list:
         exceptions: int = 0
         self._set_initialstates()
 
-        for datum in data:
+        for dataentry in data:
```
Comment (michellekunz) [inline superset/utils/geocoders.py]: data_entry

File: superset/utils/geocoders.py [hunk 4]
```diff
@@ -54,97 +71,66 @@ def _geocode_maptiler(self, data: list) -> list:
...
                     f"exception message: {e}"
                 )
             if counter == 0 and exceptions == 1:
-                message = f"Exception at the start of the geocoding process"
+                message = f"First request to geocoding API returned an Error, aborting"
                 flash(message, "error")
                 return [message, [geocoded_data, self._set_dict()]]
             if exceptions >= 2:
-                message = f"2 Consecutive Exceptions during geocoding process"
+                message = f"Geocoding API returned 2 errors in a row, aborting"
```
Comment (Sascha-Gschwind) [inline superset/utils/geocoders.py:115]: No need to fromat this string

File: superset/views/geocoding.py [hunk 5]
```diff
@@ -147,51 +146,60 @@ def geocode(self) -> Response:
...
         lon_column = request_data.get("longitudeColumnName", "lon")
         save_on_stop_geocoding = request.json.get("saveOnErrorOrInterrupt", True)
-
+        created_message = ""
         try:
             table_dto = request_data.get("datasource", models.TableDto())
-            self._check_and_create_columns(request_data)
+            column_message = self._check_and_create_columns(request_data)
+            db.session.commit()
+            created_message = f" but {column_message} please choose the overwrite option when trying again"
```
Comment (Sascha-Gschwind) [inline superset/views/geocoding.py:154]: i would rename this to message_suffix

File: superset/views/geocoding.py [hunk 6]
```diff
@@ -230,8 +238,9 @@ def _check_and_create_columns(self, request_data):
         table_name = request_data.get("datasource", "")
         table_id = table_dto.get("id", "")
         table = self._get_table_by_id(table_id)
-        lat_exists = self._does_column_name_exist(table_id, lat_column)
-        lon_exists = self._does_column_name_exist(table_id, lon_column)
+        lat_exists = self._does_column_name_exist(table, lat_column)
+        lon_exists = self._does_column_name_exist(table, lon_column)
+        column_created_message = ""
```
Comment (Sascha-Gschwind) [inline superset/views/geocoding.py:243]: I would rename this to column_message_suffix

**Review by Sascha-Gschwind — APPROVED** [review]

**Pull request merged**